### PR TITLE
[WIP] Fixes the sluggishness of live.grid

### LINKED ISSUE #28: live.grid: sluggish UI interaction in client
Interacting with live.grid is very slow on the browser side. Paste the below in Max and open a browser and mouse in the browser window:

<pre><code>
----------begin_max5_patcher----------
737.3ocsU0ziaCBD8bxuhHeN0BveRO1KqTO1CUpppJh3PbXWLDgwY2sq1+6c
3C6jTksZSUVKYaXfAdy6MC7x7YIq0Ow6SV74E+bwrYuLe1LuImgYw9yR5XO0
HY89okHEG3osFwljkgQOvLJVG+xCpF5zCVI25ctLZcOy1rSnZWY3M1vlmmkV
hfmxkKvHRJZ4hLDx8CW.+V7qnmgEy97ddvsjjkW3cZ5.PbvRu99OgSl1bC.W
K2rhqXqk9EBeDtB0HZIQi8rC7MqXVqQrdvxO1pORSQdxQExAtd6n4Q6mtmRs
p8MoqylY+Nsw99l5HejcgwDpChdwXbFF+04yO9y+804Kem5emvvR25V8Kqwn
2ViwzTBHwXxxEjxTpStofNmWmVQoz5ZvbdUJsJmRwWTByF2x0rlGZM5A0lSU
OQqRC6lTz7vaJpnSBY+djv1u+.2zKzJeT6hYHhuWabcqV56JTgtdBNwvcTZX
99jjDlAhUKDnClfb8TcYRvU8FtQMHhrOvyvV1aeV92UcSJ8WYc5N1pch1cR3
0B4dZCS0F46jM7srAo8bIh0zvU1FsLfSfqc0NnzB2GWUjqFJD5gUARP.G7.w
uqwEuCVYwHOcZRwD7T7GAs3KPh9XI06DQnzLbUYdkCQkE4TDw0h5Lg+..nQ+
n5pQ3QbUTQf1tVYePH7NCme8HLnpYtGWKbYcIo51Ctuw2b0PqpNOCm6QUdAh
V6S8HYED5sGe+fKk5GuZHVSJv9DvpZbM1CvBZQdQDfwCL1pAjI9c7JFLwMID
p9VE.Ccq4l6zx+CFFxDc2PBY.TBtvGIXDlVdSSO2yTb42EZ3Hy+E9V2tUHkS
36z6AGuOJILZ7VojyOLXp.CvOpxmsj6Rl8GYQFEinW3Q2xmpPyg4lWDbCxxx
NSCitQFciPczjubFxLQ4gVmvbitAGzF3KRkqXKZcuQuGtMNdlO3JcZ9CVcqg
sQ.b8z8KS2rdMJAnduN+Ozrk0vJ
-----------end_max5_patcher-----------
</code></pre>

There are a ton of mgraphics.fill() and mgraphics.stroke() calls in for loops that are likely the culprits.

### PR DESCRIPTION
**Pull request by catarak**
Fixes #28.

I don't think this is perfect as it is right now, but it's definitely reached a point where @fde31 or @benbrackenc74 should take a look and see how it's working. The code itself could also probably use a little refactoring, as it is pretty insane.

### CODE REVIEW

**Comment by catarak** [comment]
oops i included some random linting fixes on this too, sorry 🙃

**Comment by catarak** [comment]
@benbrackenc74 i just pushed a change which I _think_ resolves the scale issue you were seeing, but definitely confirm that it it is good 🙃

File: src/js/objects/live.grid.js [hunk 1]
```diff
@@ -1,173 +1,496 @@
...
+import ActiveFrameStore from "../stores/activeFrame.js";
 
-export default class LiveGrid extends MiraUIObject {
-	constructor(stateObj) {
-		super(stateObj);
-		this._typeForTouch = {};
-		this._lastPositionForTouch = {};
+const DIRECTION_MARGIN = 6;
+
+const CellStyle = Object.freeze({
```
Comment (fde31) [inline src/js/objects/live.grid.js]: if we want immitate immutability here, we might wanna use nested Object.freeze!?

**Comment by fde31** [comment]
Seems all good to me and performs nicely. We still do a bunch of graphics rendering for each cell I guess but at least for me that seems to perform well for me on my MBP, iPhone 6 and iPar Air. I was wondering if we could be a bit smarter with cached SVG textures (would have to reload them on resize though) and applying them in a tinted way to cells but I guess there is no need at this point. So yeah, apart from the nested Object.freeze I'm fine with merging.

**Comment by catarak** [comment]
this is _finally_ fixed!

**Comment by benbrackenc74** [comment]
Thanks, Cassie!

**Pull request merged**